cmake/kokkos_tpls.cmake

```diff
@@ -85,6 +85,11 @@ ENDIF()
 KOKKOS_IMPORT_TPL(ROCM INTERFACE)
 KOKKOS_IMPORT_TPL(LIBQUADMATH)
 
+IF (Kokkos_ENABLE_IMPL_DESUL_ATOMICS AND Kokkos_ENABLE_DESUL_ATOMICS_EXTERNAL)
+  find_package(desul_atomics REQUIRED QUIET)
+  KOKKOS_EXPORT_CMAKE_TPL(desul_atomics)
+ENDIF()
```

Comment:
On second thought, we do not need to introduce an option for this since there is native support for it
* [`CMAKE_DISABLE_FIND_PACKAGE_PackageName`](https://cmake.org/cmake/help/v3.16/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html#cmake-disable-find-package-packagename) to disable the `find_package()` call
* [`CMAKE_REQUIRE_FIND_PACKAGE_PackageName`](https://cmake.org/cmake/help/v3.23/variable/CMAKE_REQUIRE_FIND_PACKAGE_PackageName.html#cmake-require-find-package-packagename) (since CMake 3.22) to turn it into a required dependency

Reply:
It's just that I can't have the behavior I want for the CI (force finding an external library) with its `CMake` version. Since this is also what users might want, I would prefer the additional CMake variable that can be deprecated/removed once we require CMake 3.22.

Of course, I'm happy to go with the majority vote.

Reply:
I'll try to add my different views on the issue, as a system admin, as a normal user and as a developer.

system admin:
If I install libraries on a system and this installation is possibly shared (e.g. via a module system) to a cluster or just to multiple computers that are used by "common computer knowledge users", I am basically trying to take the configuration and compiling off their desks and onto my desk to help them focus on their work. In this scenario I definitely want the configuration to fail (if the output of cmake is longer than one page) or at least emit a highlighted warning as soon as I would have anything I am dependent on in order to decide if I am okay with an internal or want to externalize it. That said, if I have the option to use a package manager I can offload the hassle of thinking about this to the manager knowing it takes care of this internally (if it can).

normal user who can not offload to the sysadmin:
An error at configure time due to failing to find a lib would cause irritation and frustration that I need to take care of this. If it would be just a warning that an internal version is used i would not care but would be blind to the consequences. Even more so if it configures with just a status message. If it would give me an error but also hint to the possibility of using an internal version and tell me the configure options, I would be happy again.

kokkos developer/advanced user:
As long as I am aware of what it means when I am using the internal version I am fine with it. So having no warnings in the cmake output once I specify which option to use would be fine. For a CI I would like to get an error if something warns but this could be handled with `-Werror=dev` of cmake.

tl:dr
I would like to see an error if find_package fails. The error message should give the argument to use the internal desul for normal users that don't care together with a short explanation. If the internal is used I would still print a highlighted warning in the configure output. For the external I am fine with having no warning but just a status, as this is the "expected" behavior. I am in favor of not adding another variable and stick to the cmake commands.

Reply:
We decided to add a `CMake` variable that forces using an external installation. I suggested `Kokkos_ENABLE_DESUL_ATOMICS_EXTERNAL` here but don't care too much if someone has a better suggestion.